tests/containers/container_engine.pm

```diff
@@ -94,7 +94,7 @@ sub basic_container_tests {
 
     ## Test `--init`, i.e. the container process won't be PID 1 (to avoid zombie processes)
     # expected output: the `ps` command is not running as PID 1. The $runtime-init process shall be 1 instead.
-    validate_script_output("$runtime run --rm --init $image ps --no-headers -xo 'pid args'", sub { $_ =~ m/\s*1 .*${runtime}-init .*/ });
+    validate_script_output("$runtime run --rm --init $image ps --no-headers -xo 'pid args'", sub { $_ =~ m/\s*1 .*(${runtime}-|\/)init .*/ });
```

Comment:
This will not work in all cases. The `${runtime}-init` is not always present. See e.g. the output on 15-SP2:

```
sles15-sp2:~ # podman run --rm --init registry.opensuse.org/opensuse/tumbleweed ps --no-headers -xo 'pid args'
Trying to pull registry.opensuse.org/opensuse/tumbleweed...
Getting image source signatures
Copying blob 33f99bfc68ed done  
Copying config b2e19f9503 done  
Writing manifest to image destination
Storing signatures
    1 /dev/init -- ps --no-headers -xo pid args
    6 ps --no-headers -xo pid args
```

So, if we really want to alter the check, then let's do it in a robust way of probing for `init`. We can re-add the actual check for the `ps` command in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/18844 after this PR is merged.

```suggestion
    validate_script_output("$runtime run --rm --init $image ps --no-headers -xo 'pid args'", sub { $_ =~ m/\s*1 .*init .*/ });
```

Also this change invalidates the comment above - the `ps` command is not probed anymore, so let's please remove that part of the comment.

Reply:
Let me know if you agree to leave the names matching more controlled, instead of match `*init`.

Reply:
I now understand what you want to do. Yes this makes sense, but we need to add a better comment. I suggest something like

```
# Ensure PID 1 has either the $runtime-init (e.g. podman-init) or /init (e.g. `/dev/init) suffix
```

Reply:
I can confirm that for all these 3 cases Maurizios suggestion should match.
Although the idea behind the slash is difficult to see, given that it is always matched by the whole expression, but in the case of a preceeding runtime string by following part: ".*" instead of the expression in parenthesis, which matches the slash only if there is no runtime prefix, but allows the part in the parenthesis to be mandatory to match the expression and to prevent e.g.: "garbage-init" to match.

Reply:
Agreed, but we need to leave comments for our future selves when a poor reviewer (TM) needs to debug an issue here and find themselves in the situation of needing to understand this regex.

Reply:
Yeah maybe something like the following might give a sufficient hint I guess:
```Perl
# Ensures that only "$runtime-init" or "init" can match for the executable.
```

Reply:
There could be other ways to express the meaning, but the form I used here is a compromise of readability(for reviews and debugging) and matching constraints.

Reply:
The comment does not cover the case of "/init" and that was the thing that confused me - because there is a discrepancy between the written comment and the code.

Plus the first part of the comment is not being checked for, so it should be removed.

Reply:
yes, new vcomment will be:
```
## Test `--init`, option
# Ensure PID 1 has either the $runtime-init (e.g. podman-init) OR /init (e.g. `/dev/init) suffix
```

Reply:
Perfect :+1:

Reply:
Please leave the `## Test `--init`, i.e. the container process won't be PID 1 (to avoid zombie processes)` though, that is crucial information about what the `--init` test is supposed to be doing.

Reply:
aggh! just pushed... will redo.

Reply:
Adding, however this sentence refers to the next coming test of `ps` process and not to this acutal on `init`.